### PR TITLE
NIFI-8772 - address Maven build warnings

### PR DESCRIPTION
**Pull request by greyp9**
Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

Fix Maven POM parent/child inconsistencies.  Address Maven build warnings. 

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [X] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [X] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [X] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-graph-bundle/nifi-graph-processors/pom.xml [hunk 1]
```diff
@@ -84,11 +84,6 @@
             <version>1.14.0-SNAPSHOT</version>
             <scope>compile</scope>
         </dependency>
-        <dependency>
```
Comment (greyp9) [inline nifi-nar-bundles/nifi-graph-bundle/nifi-graph-processors/pom.xml:87]: [WARNING] Some problems were encountered while building the effective model for org.apache.nifi:nifi-graph-processors:jar:1.14.0-SNAPSHOT
[WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: org.apache.nifi:nifi-record-serialization-service-api:jar -> version (?) vs 1.14.0-SNAPSHOT @ line 98, column 21

File: nifi-registry/nifi-registry-core/nifi-registry-web-api/pom.xml [hunk 2]
```diff
@@ -467,11 +467,6 @@
             <version>3.2.1</version>
             <scope>test</scope>
         </dependency>
-        <dependency>
```
Comment (greyp9) [inline nifi-registry/nifi-registry-core/nifi-registry-web-api/pom.xml:470]: [WARNING] Some problems were encountered while building the effective model for org.apache.nifi.registry:nifi-registry-web-api:war:1.14.0-SNAPSHOT
[WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: org.spockframework:spock-core:jar -> duplicate declaration of version (?) @ line 480, column 21

File: nifi-external/nifi-example-bundle/nifi-nifi-example-nar/pom.xml [hunk 3]
```diff
@@ -22,7 +22,7 @@
         <version>1.14.0-SNAPSHOT</version>
     </parent>
 
-    <artifactId>nifi-example-nar</artifactId>
+    <artifactId>nifi-nifi-example-nar</artifactId>
```
Comment (greyp9) [inline nifi-external/nifi-example-bundle/nifi-nifi-example-nar/pom.xml:25]: https://github.com/apache/nifi/blob/main/nifi-external/nifi-example-bundle/pom.xml#L30

File: nifi-nar-bundles/nifi-standard-services/nifi-ssl-context-bundle/pom.xml [hunk 4]
```diff
@@ -23,6 +23,6 @@
     <packaging>pom</packaging>
     <modules>
         <module>nifi-ssl-context-service</module>
-        <module>nifi-ssl-context-nar</module>
+        <module>nifi-ssl-context-service-nar</module>
```
Comment (greyp9) [inline nifi-nar-bundles/nifi-standard-services/nifi-ssl-context-bundle/pom.xml:26]: https://github.com/apache/nifi/blob/main/nifi-nar-bundles/nifi-standard-services/nifi-ssl-context-bundle/nifi-ssl-context-nar/pom.xml#L22

**Review by exceptionfactory — APPROVED** [review]
Thanks for correcting these issues @greyp9, looks good! +1 Merging.

**Pull request closed without merging**